src/explore-education-statistics-common/src/modules/charts/components/VerticalBarBlock.tsx

```diff
@@ -74,21 +76,19 @@ const VerticalBarBlock = ({
   const yAxisWidth = parseNumber(axes.minor.size);
   const xAxisHeight = parseNumber(axes.major.size);
 
-  const categoryDataSetConfigurations = getCategoryDataSetConfigurations(
+  const dataSetCategoryConfigs = getDataSetCategoryConfigs(
     dataSetCategories,
-    axes.major,
+    legend.items,
     meta,
   );
 
-  const minorAxisDecimals = getMinorAxisDecimalPlaces(
-    categoryDataSetConfigurations,
-  );
+  const minorAxisDecimals = getMinorAxisDecimalPlaces(dataSetCategoryConfigs);
 
   return (
     <ChartContainer
       height={height || 300}
       legend={legendProps}
-      legendPosition={legend}
+      legendPosition={legend.position}
```

Comment:
The `legend?.position` below on line 139 suggests that legend could be null.  Does the same check need applying here too?

Reply:
Woops, just a typo on `legend.position`. Legend should be defined for `VerticalBarBlock`